src/converterStrategy/baseConverterStrategy.ts

```diff
@@ -84,70 +84,73 @@ module powerbi.extensibility.visual.converterStrategy {
             let categoryGradientStartBaseColorIdentities: BaseColorIdentity[] = [];
             let categoryGradientEndBaseColorIdentities: BaseColorIdentity[] = [];
             let categoryItemsCount: Array<IFilteredValueGroups[]> = [];
-            this.dataView.categories[0].values.forEach( (category: PrimitiveValue, index: number) => {
-                // gradiend start color
-                let mappedItems: IFilteredValueGroups[] = valueGroups.map( group => {
-                    return <IFilteredValueGroups>{
-                        gr: group,
-                        categoryValue: group.values[0].values[index],
-                        categoryIndex: index,
-                        category: category,
-                        identity: group.identity
-                    };
-                }).filter(v => v.categoryValue !== null);
 
-                categoryItemsCount[index] = mappedItems;
+            if (colorGradient) {
+                this.dataView.categories[0].values.forEach( (category: PrimitiveValue, index: number) => {
+                    // gradiend start color
+                    let mappedItems: IFilteredValueGroups[] = valueGroups.map( group => {
```

Comment:
It'd be better to use forEach and push IFilteredValueGroups to mappedItems if and only if categoryValue !== null.
This improvement might work faster than the current implementation.
What is your point of view?